launch_ros/launch_ros/substitutions/find_package.py

```diff
@@ -28,27 +30,28 @@
 from launch.utilities import perform_substitutions
 
 
-@expose_substitution('find-pkg')
 class FindPackage(Substitution):
     """
-    Substitution that tries to locate the package prefix of a ROS package.
-
-    The ROS package is located using ament_index_python.
+    Abstract base class for substitutions involving finding a package.
 
-    :raise: ament_index_python.packages.PackageNotFoundError when package is
-        not found during substitution.
+    Subclasses should implement the find() method.
     """
 
-    def __init__(self, package: SomeSubstitutionsType) -> None:
+    def __init__(
+        self,
+        package: SomeSubstitutionsType,
+        find_package_func: Callable[[str], str] = get_package_prefix
+    ) -> None:
         """Constructor."""
         super().__init__()
         self.__package = normalize_to_list_of_substitutions(package)
+        self.__find_package_func = find_package_func
```

Comment:
@jacobperron why keep the `self.__find_package_func` attribute?

Reply:
Left-over by mistake. Removed: 1cb949e